src/utils/conversions.ts

```diff
@@ -12,20 +13,29 @@ export function satsToUsd(
         const btc = MutinyWallet.convert_sats_to_btc(
             BigInt(Math.floor(amount))
         );
-        const usd = btc * price;
-
-        if (formatted) {
-            return usd.toLocaleString("en-US", {
+        const fiatPrice = btc * price;
+        if (!formatted && fiatPrice === 0) {
+            return fiatPrice.toFixed(0);
+        } else if (fiat === "BTC" && fiatPrice !== 0) {
+            return fiatPrice.toFixed(8);
+        } else if (formatted && !fiat?.match("BTC|CHF")) {
+            return fiatPrice.toLocaleString(navigator.languages[0] || "en-US", {
                 style: "currency",
-                currency: "USD"
+                currency: fiat || "USD",
+                currencyDisplay: "narrowSymbol",
+                maximumFractionDigits:
+                    fiat?.match("KRW|JPY") || fiatPrice === 0 ? 0 : 2
             });
         } else {
             // Some float fighting shenaningans
-            const roundedUsd = Math.round(usd);
-            if (roundedUsd * 100 === Math.round(usd * 100)) {
-                return usd.toFixed(0);
+            const roundedFiat = Math.round(fiatPrice);
+            if (
+                roundedFiat * 100 === Math.round(fiatPrice * 100) ||
+                fiat?.match("KRW|JPY")
```

Comment:
need to discuss the need for these hardcoded values doesn't seem like it will scale well to other currencies but also I don't want this logic any more complicated or my poor brain will fry!